lti_consumer/api.py

```diff
@@ -88,6 +88,47 @@ def _get_lti_config(config_id=None, block=None):
     return lti_config
 
 
+def get_or_update_lti_config(block):
+    """
+    Gathers all the LTI 1.3 related values from the passed block and stores
+    it in the corresponding LtiConfiguration.
+
+    Arguments:
+        block (XBlock): the block whos configuration should be updated in the db
+
+    Returns:
+        the updated LtiConfiguration object
+    """
+    conf = _get_lti_config(block=block)
+
+    if block.config_type == "external":
+        return conf
+
+    if block.lti_version == "lti_1p3":
+        # Transfer the config to DB so that the django view can initialize the
+        # LTI1P3Consumer without having to load the XBlock
```

Comment:
@tecoholic What's the reasoning behind ditching `CONFIG_ON_XBLOCK` in favor of `CONFIG_ON_DB` and copying over all the configuration? As far as I understand, this is not really needed to achieve the reusability we're looking for and adds extra complexities to syncing data between the block and the model.

@tecoholic @tecoholic @MichaelRoytman  Was there any discussion on the previous PRs about this?

Reply:
CCing @Zacharis278 on this discussion too ^

Reply:
Something that was on my mind while looking for this was if we're moving from configuration in modulestore to the database it does seem like something that should have its own ADR to explain the reasoning/benefit.

Maybe it would be a good idea to split that into its own feature? What if we just synced the parameters in memory here and didn't save back to the DB yet?

Reply:
@giovannicimolin @Zacharis278 Thank you for the comments. I have been thinking about this in my head for about a 2 weeks now. I think I might have over-engineered the "LTI Launch without XBlock" here. It happened when I was testing backward compatibility for the XBlock's handlers. The flow ends up something like this:

1. The XBlocks's handler is called
2. The XBlock handler forwards the request to the Django View 
3. The Django view loads the config from XBlock, thereby creating another instance of the XBlock withing it's context

This seemed too convoluted at that time and I couldn't reason out the context when Unit tests failed. So syncing the data to the DB, seemed like a clearer path -  *The Django View always loads the config from DB* - this seemed like a much simpler way to reason about.

But after the changes introduced in #260, it has become somewhat of a code smell. The implicit migration of data to the DB from XBlock is not aligned with the explicit WaffleFlag approach. I also have come to realize that this has just shifted the complexity & interdependency between XBlock & Django View to XBlock & Django Model. Which in itself might not be bad, but definitely seems like it needs some clarity along the lines of the @Zacharis278's questions.

I will drop this from the PR and let you know.